test/e2e/rule_test.go

```diff
@@ -556,6 +560,141 @@ func TestRule(t *testing.T) {
+	t.Cleanup(e2ethanos.CleanScenario(t, s))
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	t.Cleanup(cancel)
+
+	// Prepare work dirs.
+	rulesSubDir := filepath.Join("rules")
+	rulesPath := filepath.Join(s.SharedDir(), rulesSubDir)
+	testutil.Ok(t, os.MkdirAll(rulesPath, os.ModePerm))
+	createRuleFile(t, filepath.Join(rulesPath, fmt.Sprintf("rules-0.yaml")), testAlertRuleRecordAbsentMetric)
+	amTargetsSubDir := filepath.Join("rules_am_targets")
+	testutil.Ok(t, os.MkdirAll(filepath.Join(s.SharedDir(), amTargetsSubDir), os.ModePerm))
+	queryTargetsSubDir := filepath.Join("rules_query_targets")
+	testutil.Ok(t, os.MkdirAll(filepath.Join(s.SharedDir(), queryTargetsSubDir), os.ModePerm))
+
+
+	am, err := e2ethanos.NewAlertmanager(s.SharedDir(), "1")
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(am))
+
+	receiver, err := e2ethanos.NewReceiver(s.SharedDir(), s.NetworkName(), "1", 1)
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(receiver))
+	rwURL, err := url.Parse(e2ethanos.RemoteWriteEndpoint(receiver.NetworkEndpoint(8081)))
+	testutil.Ok(t, err)
+
+	querier, err := e2ethanos.NewQuerier(s.SharedDir(), "1", []string{receiver.GRPCNetworkEndpoint()}, nil, nil, nil, nil, nil, "", "")
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(querier))
+
+	r, err := e2ethanos.NewRuler(s.SharedDir(), "1", rulesSubDir, []alert.AlertmanagerConfig{
+		{
+			EndpointsConfig: http_util.EndpointsConfig{
+				FileSDConfigs: []http_util.FileSDConfig{
+					{
+						// FileSD which will be used to register discover dynamically am1.
+						Files:           []string{filepath.Join(e2e.ContainerSharedDir, amTargetsSubDir, "*.yaml")},
+						RefreshInterval: model.Duration(time.Second),
+					},
+				},
+				StaticAddresses: []string{
+					am.NetworkHTTPEndpoint(),
+				},
+				Scheme: "http",
+			},
+			Timeout:    model.Duration(time.Second),
+			APIVersion: alert.APIv1,
+		},
+	}, []query.Config{
+		{
+			EndpointsConfig: http_util.EndpointsConfig{
+				// We test Statically Addressed queries in other tests. Focus on FileSD here.
+				FileSDConfigs: []http_util.FileSDConfig{
+					{
+						// FileSD which will be used to register discover dynamically q.
+						Files:           []string{filepath.Join(e2e.ContainerSharedDir, queryTargetsSubDir, "*.yaml")},
+						RefreshInterval: model.Duration(time.Second),
+					},
+				},
+				Scheme: "http",
+			},
+		},
+	}, true,  remotewrite.Config{
+		Name: "ruler-rw-receivers",
+		RemoteStore: &config.RemoteWriteConfig{
+			URL: &commoncfg.URL{URL: rwURL},
+			Name: "thanos-receiver",
+		},
+	})
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(r))
+
+	writeTargets(t, filepath.Join(s.SharedDir(), queryTargetsSubDir, "targets.yaml"), querier.NetworkHTTPEndpoint())
+	writeTargets(t, filepath.Join(s.SharedDir(), amTargetsSubDir, "targets.yaml"), am.NetworkHTTPEndpoint())
+
+	t.Run("inject samples into receiver to reset its StoreAPI MinTime", func(t *testing.T) {
```

Comment:
This works great, and nice way of generating data. Some thoughts:

* There is no need into creating Prometheus with all extra functionalities. We know that CI GH actions have e limited CPU  time, so we don't want to run extra container if not needed. To generate data, we can quickly craft small remote write request using literally what @metalmatze used in his PR. Just make sure you use the local endpoint, not network endpoint.
* We don't want to reset some min time, we literally want generate, data so can we rename this subtest accordingly (and comment)

OR alternatively we can remove all of this, because our expression does not use any Store API data right now (unless you want to change that). Might simplify our test!